Deprecate mixing positional and keyword args for legend(handles, labels) (#27175)
matplotlib/matplotlib: doc/api/next_api_changes/deprecations/27175-AL.rst

Mixing positional and keyword arguments for ``legend`` handles and labels
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> This was previously a plain warning, but is now deprecated.  If passing

Reviewer: ```suggestion
This previously only raised a warning, but is now formally deprecated.  If passing
```

Author: sure, reworded.
